### PR TITLE
fix: lazy mapped keys with functions

### PR DESCRIPTION
**Pull request by tris203**
keys set lazy that returned functions returned the function not a string

### CODE REVIEW

File: lua/hawtkeys/ts.lua [hunk 1]
```diff
@@ -378,7 +378,7 @@ local function get_keymaps_from_lazy()
                     for _, key in pairs(v._.handlers.keys) do
                         local map = {
                             lhs = key.lhs,
-                            rhs = key.rhs,
+                            rhs = tostring(key.rhs),
```
Comment (willothy) [inline lua/hawtkeys/ts.lua:390]: could also do a check for function values and check `debug.getinfo`

Comment (tris203) [reply lua/hawtkeys/ts.lua:390]: image
done that, will print using the file and line where it is defined now

File: lua/hawtkeys/ts.lua [hunk 2]
```diff
@@ -376,6 +376,15 @@ local function get_keymaps_from_lazy()
             for _, v in ipairs(lazy) do
                 if v and v._ and v._.handlers and v._.handlers.keys then
                     for _, key in pairs(v._.handlers.keys) do
+                        if type(key.rhs) == "table" then
+                            key.rhs = tostring(key.rhs)
+                        elseif type(key.rhs) == "function" then
+                            local debugInfo =
+                                debug.getinfo(key.rhs --[[@as function]], "S")
```
Comment (willothy) [inline lua/hawtkeys/ts.lua]:
```suggestion
                                debug.getinfo(key.rhs --[[@as fun()]], "S")
```

Comment (willothy) [reply lua/hawtkeys/ts.lua]: afaik this is the correct way to write function types

**Review by willothy — COMMENTED** [review]
just the type annotation then lgtm!

**Review by willothy — APPROVED** [review]
lgtm!

**Pull request merged**